src/test/scala/apps/cameraPipeCheck.scala

```diff
@@ -105,12 +112,53 @@ float clamp_f32(float v, float l, float h) {
 #define pow_f32 powf
 """
 
+  val insertCopyAtEnd: Strategy[Rise] =
+  // if...
+    `try`(
+      // ...there is no computation in the AST...
+      predicate.not(oncetd(isComputation)) `;`
+        // ...insert copy: .
+        oncetd(
+          // ...skip all (Dep-)Lambdas...
+          // ( <+ can be read as `or` ) -> maybe introduce `||` notation?
+          predicate.not(isDepLambda <+ isLambda) `;`
+            // ...insert Id() and transform it to an actual copy
+            idAfter `;` normalize.apply(liftId) `;` oncetd(idToCopy)
```

Comment:
It seems like you could directly use `fun(x => x)` instead of an id primitive here :thinking:

Reply:
I think we should also actually talk about "writing to memory" instead of "copying". We make the decision to write an expression in memory, not really to copy it from one memory to another.